src/udiskslinuxpartitiontable.c

```diff
@@ -691,41 +364,11 @@ udisks_linux_partition_table_handle_create_partition (UDisksPartitionTable   *ta
       g_clear_object (&partition_object);
       goto out;
     }
-  escaped_partition_device = udisks_daemon_util_escape_and_quote (udisks_block_get_device (partition_block));
 
-  /* set partition type */
-  if (strlen (type) == 0)
-      set_type = FALSE;
-
-  if (set_type)
-    {
-      partition = udisks_object_get_partition (partition_object);
-      if (!partition)
-        {
-          g_dbus_method_invocation_return_error (invocation,
-                                                 UDISKS_ERROR,
-                                                 UDISKS_ERROR_FAILED,
-                                                 "Error getting newly created partition");
-          goto out;
-        }
-
-      if (! udisks_linux_partition_set_type_sync (UDISKS_LINUX_PARTITION (partition),
-                                                  type,
-                                                  caller_uid,
-                                                  NULL,
-                                                  &error))
-        {
-          g_prefix_error (&error, "Error setting type for newly created partition: ");
-          g_dbus_method_invocation_take_error (invocation, error);
-          g_clear_object (&partition);
-          goto out;
-        }
-
-      g_clear_object (&partition);
-    }
+  escaped_partition_device = udisks_daemon_util_escape_and_quote (udisks_block_get_device (partition_block));
```

Comment:
This should call ``dup_device()`` too (and the returned value should be free'd because ``escape_and_quote()`` allocates new memory, AFAICT.